### DISCUSSION TITLE
Top languages card is empty

### DISCUSSION THREAD

**Discussion by fqrtg14** [anuraghazra/github-readme-stats#1910] [Q&A] [answered]
**Describe the bug**
I have created my Vercel instance and am using it but my languages card does not show anything.

**Expected behaviour**
The languages card displays something.

**Screenshots / Live demo link (paste the github-readme-stats link as markdown image)**
![image](https://user-images.githubusercontent.com/20853284/182013871-3ea4ee1b-46f8-4c9a-aad3-698101d72d2d.png)
https://github-readme-stats-fqrtg14.vercel.app/api/top-langs/?username=fqrtg14&&layout=compact

**Comment by rickstaa** [comment, marked as answer]
@fqrtg14  Thanks for your issue. I checked your account, and it looks like the results shown on the card are correct. You can see this in action by using the following query in the [graphql explorer](https://docs.github.com/en/graphql/overview/explorer):

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    # fetch only owner repos & not forks
    repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

You can then use the following in the query variables field.

```graphql
{"login": "fqrtg14"}
```

There are only two repositories that show up, namely `hello-world` and `fqrtg14`. These repositories, however, do not have any language info attached to them. This is probably because they only contain languages that are not regarded by [linguist ](https://github.com/github/linguist)as languages (see https://github.com/github/linguist/issues/5951 for more info). Further, please be aware that we currently do not fetch data from forks only from parent repositories and that the public Vercel instance does not have access to your private repositories (see [the FAQ](https://github.com/anuraghazra/github-readme-stats/discussions/1770) for more info).

**Reply by fqrtg14** [reply]
Thank for your answer. Then, is there way to access to my private repositories? I wonder if there are other options besides the public Vercel intance.

**Reply by rickstaa** [reply]
@fqrtg14 Great! Please accept my answer as the accepted answer if it solves your problem so we can close this question 😄. Since we can't access your private repositories, you have to deploy a Vercel instance yourself for the stats of these private repositories to show up on your card. You can find a guide on how this is done in the [README](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance). Feel free to let me know if you have any problems deploying your own instance.

**Reply by fqrtg14** [reply]
I refered codeSTACKr's video and deployed my Vercel instance. I think that deploying is completed without problems but it's not working. What should I check?

**Reply by rickstaa** [reply]
@fqrtg14 No problem. If you created a [personal access token](https://github.com/settings/tokens/new) (PAT) with repository access

![image](https://user-images.githubusercontent.com/17570430/182026645-3625b0bd-74c0-4805-991f-d90633839520.png)

and added it as a `PAT_1` Vercel environmental variable

![image](https://user-images.githubusercontent.com/17570430/182026664-c37b2496-56ec-4ca4-967e-4c7a1dc4b87f.png)

your new card should be available under the URL Vercel gives for your deployment. You can find this URL under the deployments tab in the Vercel dashboard

![image](https://user-images.githubusercontent.com/17570430/182026733-2a519a59-d391-433c-a3ef-5b3049843aa5.png).

**Reply by fqrtg14** [reply]
The problem was the wrong URL.. Now it works fine. Thank you so much!
But in my case, I was able to find the URL in the overview tab instead of the deployments tab..
I tried using the URL in the deployments tab but it didn't work.
Do you know why?

**Reply by fqrtg14** [reply]
Sorry to confuse you. I tried a little more. If I enter both the URL in the deployments tab and the URL in the overview tab into a web browser, the languages ​​card appears properly. It works well with or without compact layout. So it doesn't seem to be a URL issue.
But in the readme file, if I write &layout=compact, the card looks like it's empty, and if I don't, it shows properly. 
Is it a bug? Or am I doing something wrong?

[![Top Langs](https://github-readme-stats-nine-brown.vercel.app/api/top-langs/?username=fqrtg14)](https://github.com/anuraghazra/github-readme-stats)
[![Top Langs](https://github-readme-stats-nine-brown.vercel.app/api/top-langs/?username=fqrtg14&layout=compact)](https://github.com/anuraghazra/github-readme-stats)

**Reply by rickstaa** [reply]
@fqrtg14 Maybe it was a caching issue. Both options seem to work now.

```md
[![Top Langs](https://github-readme-stats-nine-brown.vercel.app/api/top-langs/?username=fqrtg14)](https://github.com/anuraghazra/github-readme-stats)
```

[![Top Langs](https://github-readme-stats-nine-brown.vercel.app/api/top-langs/?username=fqrtg14)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Top Langs](https://github-readme-stats-nine-brown.vercel.app/api/top-langs/?username=fqrtg14&layout=compact)](https://github.com/anuraghazra/github-readme-stats)
```

[![Top Langs](https://github-readme-stats-nine-brown.vercel.app/api/top-langs/?username=fqrtg14&layout=compact)](https://github.com/anuraghazra/github-readme-stats)

**Reply by fqrtg14** [reply]
I think so too.. Thank for your help. :)

**Comment by emjose** [comment]
Hello, thank you for this discussion! 

@rickstaa 

- I am having the same issue with a top languages card that is blank.
- The card started appearing blank on Wednesday, August 24th, 2022.
- I've noticed that some GitHub users I follow also have blank top languages cards, which were not previously blank.
- I forked the repo, enabled repo permissions in the personal access token, and deployed my Vercel instance.
- The Github stats card works correctly with my Vercel instance.
- However, the top languages card is still blank. The 3 domains generated by Vercel were all tested.

# 

#### My Github stats (with private repos included) are appearing correctly with my Vercel instance.
![Github Stats Card](https://github-readme-stats-flame-phi.vercel.app/api?username=emjose&count_private=true)

**https://github-readme-stats-flame-phi.vercel.app/api?username=emjose&count_private=true**

```
https://github-readme-stats-flame-phi.vercel.app/api?username=emjose&count_private=true
```

# 

#### My top languages card is still blank.
![Github Stats Card](https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=emjose)

**https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=emjose**

```
https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=emjose
```

# 

- Are there any steps I have overlooked?
- Please let me know if this should be a new issue entirely.

**Reply by rickstaa** [reply]
@emjose I checked your card, and I don't think something is going wrong on our end. I can get my language stats when I use your vercel instance.

```markdown
[![Top Langs](https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=rickstaa)](https://github.com/anuraghazra/github-readme-stats)
```

[![Top Langs](https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=rickstaa)](https://github.com/anuraghazra/github-readme-stats)

When I check your stats in the [graphql explorer](https://docs.github.com/en/graphql/overview/explorer), I get the following results:

```graphql
{
  "data": {
    "user": {
      "repositories": {
        "nodes": [
          {
            "name": "vedge",
            "languages": {
              "edges": []
            }
          },
          {
            "name": "learn-co-sandbox",
            "languages": {
              "edges": []
            }
          },
          {
            "name": "solid-telegram",
            "languages": {
              "edges": []
            }
          },
          {
            "name": "glowing-train",
            "languages": {
              "edges": []
            }
          }
        ]
      }
    }
  }
}
```

This shows that you only have repositories that contain languages that are not regarded by [linguist ](https://github.com/github/linguist) as languages (see https://github.com/github/linguist/issues/5951 for more info). I, however, do not have permission to check your private repositories, so I don't know if these contain language data we can use. It could be that something went wrong while deploying your Vercel instance. Maybe update your fork to the latest master branch and check whether the PAT is correct. 🤔

**Reply by emjose** [reply]
Hi @rickstaa , thank you for your response. I updated my PAT and the problem still persists.

My private repos consist of HTML, JavaScript, CSS, Ruby, and Markdown. I'm perplexed because the card appeared correctly yesterday morning, and then became blank in the evening. The only work I did in the interim was markdown in 1 repo.

What's also puzzling is when I test the GitHub usernames of classmates who were in my coding bootcamp cohort, their top languages cards are also blank, but their GitHub stats are fine (tested with both the public Vercel deployment and my own Vercel instance). 

Could this be a cache issue?

**Reply by rickstaa** [reply]
Very strange. Can you maybe show me what you get when you use the following query in the [graphql explorer](https://docs.github.com/en/graphql/overview/explorer):

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    # fetch only owner repos & not forks
    repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

with the following variables:

```graphql
{"login": "emjose"}
```

**Reply by emjose** [reply]
Here is what is returned:

```graphql
{
  "data": {
    "user": {
      "repositories": {
        "nodes": [
          {
            "name": "boris-tarantula",
            "languages": {
              "edges": []
            }
          },
          {
            "name": "vedge",
            "languages": {
              "edges": []
            }
          },
          {
            "name": "learn-co-sandbox",
            "languages": {
              "edges": []
            }
          },
          {
            "name": "solid-telegram",
            "languages": {
              "edges": []
            }
          },
          {
            "name": "glowing-train",
            "languages": {
              "edges": []
            }
          }
        ]
      }
    }
  }
}
```

**Reply by rickstaa** [reply]
Then the empty card that your Vercel instance shows is correct since that is the data we use to generate the card. Are you logged in with your GitHub user account? 🤔

Are certain that you have repositories that contain code that is regarded by [linguist ](https://github.com/github/linguist) as languages (see https://github.com/github/linguist/issues/5951 for more info)? If you are sure, I think it is best to contact GitHub support and let them know that there is a problem with the GraphQL API (see https://support.github.com/).

**Reply by rickstaa** [reply]
You can also deploy the project locally, place a breakpoint and check what data is received from the GraphQl API. See https://github.com/anuraghazra/github-readme-stats/blob/master/CONTRIBUTING.md#local-development and https://github.com/vercel/vercel/issues/2864#issuecomment-963533339 for more information on how to do this.

**Reply by emjose** [reply]
Yes, I'm logged in with my GitHub user account, and my repos all have languages recognized by Linguist.

Thank you for your recommendation, I will get in touch with GitHub Support regarding the GraphQL API.

**Reply by rickstaa** [reply]
Very strange! I have never experienced the problem you are experiencing. Please let me know what they say 🤓.

**Reply by emjose** [reply]
Thank you, I will!

**Reply by emjose** [reply]
@rickstaa The most used languages card is now working again! 

![Top Languages](https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=emjose)

[https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=emjose](https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=emjose)
```
https://github-readme-stats-flame-phi.vercel.app/api/top-langs/?username=emjose
```

I haven't received a response yet from my GitHub Support ticket, but the GraphQL issue must have cleared up on their end. 

Thank you again for all your help!
